src/librustc/ty/sty.rs

```diff
@@ -2299,20 +2298,7 @@ impl<'tcx> Const<'tcx> {
         ty: Ty<'tcx>,
     ) -> Option<u128> {
         assert_eq!(self.ty, ty);
-        // This is purely an optimization -- layout_of is a pretty expensive operation,
```

Comment:
is this (iirc recently added) optimisation being replaced by your change or was this removal "unintentional"

Reply:
It's intentional. The commit message says:
> PR #64673 introduced some special-case handling of scalar types in
`Const::try_eval_bits()`. This handling is now moved out of that function into
the new `IntRange::integral_size_and_signed_bias` function.